lib/async_hooks.js

```diff
@@ -60,19 +60,11 @@ async_wrap.setupHooks({ init,
                         destroy: emitDestroyN });
 
 // Used to fatally abort the process if a callback throws.
-function fatalError(e) {
```

Comment:
>  if it detects stack corruption that can likely result from the user try catching/Promise wrapping the call

I see, why this is not an issue when using `MakeCallback` from C++? That will run the code:

```c++
      ClearFatalExceptionHandlers(env());
      FatalException(env()->isolate(), try_catch);
      return Local<Value>();
```